internal/graph/check.go

```diff
@@ -197,6 +199,10 @@ func union(ctx context.Context, concurrencyLimit uint32, handlers ...CheckHandle
 		}
```

Comment:
Want to verify a behaviour change.  Let's say that there is an error (i.e., err != nil) and then we receive ctx.Done(), we will not return the err.  Instead, we will return ctx.Err().  Is this expected or we want to return err if ctx.Err() is nil.

Reply:
If we receive the context cancellation (and thus receive on the `ctx.Done()` channel), and if this happens first, then we'll return `ctx.Err()`. This behavior is expected, yes.

If we encounter a case where `err != nil`, and the context is not yet cancelled, then we'll surface that error after we've determined that there is no `allowed=true` outcome. If the context is cancelled while waiting for the `allowed=true` outcome, then the statement above ☝️ applies, because the context cancellation wins. 

In both cases it seems the behavior makes sense.